util/ranger/main_test.go

```diff
@@ -14,17 +14,47 @@
 package ranger_test
 
 import (
+	"flag"
+	"fmt"
+	"os"
 	"testing"
 
+	"github.com/pingcap/tidb/testkit/testdata"
 	"github.com/pingcap/tidb/util/testbridge"
 	"go.uber.org/goleak"
 )
 
+var testData testdata.TestData
```

Comment:
since here use shared testData, the tests process order should matters.
1. if t.Parallel() should remove too?
2. if we need use testsuite of testify to keep the process order is correct? as i know the tests order is random by default. (https://stackoverflow.com/questions/31201858/how-to-run-golang-tests-sequentially)

Reply:
No. Each test modifies different part of testdata.

Reply:
No. Each test uses their own part. I did investigate these things, no need to use suite.

Reply:
get it, thanks.